frontend/chartview.cpp

```diff
@@ -101,6 +115,67 @@ void ChartView::mouseMoveEvent(QMouseEvent *event)
         lastMousePos_ = event->pos();
         event->accept();
     }
+    else if (event->buttons() & Qt::LeftButton)
+    {
+        auto x = (event->pos()).x();
+        auto y = (event->pos()).y();
+
+        auto xVal = chart()->mapToValue(event->pos()).x();
```

Comment:
A sprinkling of one-line comments throughout this function wouldn't be a bad idea.